frontend/src/modules/alarms/AlarmsPage.tsx

```diff
@@ -330,13 +331,11 @@ export const AlarmsPage = (): JSX.Element => {
   const handleConfirm = () => {
     setConfirmOpen(false);
     applyChanges();
-    setIsDisabled(true);
   };
 
   const handleDiscardConfirm = () => {
     setDiscardOpen(false);
-    setAlarmLimitsRequestDraft(alarmLimits);
-    setIsDisabled(true);
+    if (alarmLimits !== null) setAlarmLimitsRequestDraft(alarmLimits);
```

Comment:
Shouldn't we be resetting `store.controller.alarmLimits.draft` from `store.controller.alarmLimits.request`, rather than from `store.controller.alarmLimits.current`?